src/argilla/utils/modeling.py

```diff
@@ -0,0 +1,241 @@
+                prop = TermsMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators, values=["True", "False"])
+            elif dtype == "int64":
+                prop = IntegerMetadataProperty(
+                    name=name, title=title, visible_for_annotators=self.visible_for_annotators
+                )
+            elif dtype == "float64":
+                prop = FloatMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
+            else:
+                print(f"Unhandled data type for column {col}: {dtype}")
+                continue
+            properties.append(prop)
+        return properties
+
+    def _add_text_descriptives_to_metadata(
+        self, records: List[Union[FeedbackRecord, RemoteFeedbackRecord]], df: pd.DataFrame
+    ) -> List[Union[FeedbackRecord, RemoteFeedbackRecord]]:
+        """
+        Add the text descriptives metrics extracted previously as metadata
+        to a list of FeedbackDataset records.
+
+        Args:
+            records (List[Union[FeedbackRecord, RemoteFeedbackRecord]]): A list of FeedbackDataset or RemoteFeedbackDataset records.
+            df (pd.DataFrame): The text descriptives dataframe.
+
+        Returns:
+            List[Union[FeedbackRecord, RemoteFeedbackRecord]]: A list of FeedbackDataset or RemoteFeedbackDataset records with extracted metrics added as metadata.
+        """
+        # Loop through the records and add the metrics to the metadata
+        modified_records = []
+        # Make tqdm optional
+        for record, metrics in tqdm(
+            zip(records, df.to_dict("records")),
+            total=len(records),
+            disable=not self.progress,
+            desc="Getting text descriptives...",
+        ):
+            record.metadata.update(metrics)
+            modified_records.append(record)
+        return modified_records
+
+    def update_dataset(
+        self, dataset: Union[FeedbackDataset, RemoteFeedbackDataset]
+    ) -> Union[FeedbackDataset, RemoteFeedbackDataset]:
+        """
+        Extract text descriptives metrics from a records in a FeedbackDataset
+        or RemoteFeedbackDataset, add them as metadata to the records and
+        return the updated dataset.
+
+        Args:
+            dataset (Union[FeedbackDataset, RemoteFeedbackDataset]): A FeedbackDataset or RemoteFeedbackDataset.
+
+        Returns:
+            Union[FeedbackDataset, RemoteFeedbackDataset]: A FeedbackDataset or RemoteFeedbackDataset with text descriptives metrics added as metadata.
+        """
+        if isinstance(dataset, FeedbackDataset):
+            records = dataset.records
+        elif isinstance(dataset, RemoteFeedbackDataset):
+            records = dataset.pull().records
+        else:
+            raise ValueError(f"Provided `dataset` is of `type={type(dataset)}` while only `type=FeedbackDataset` or `type=RemoteFeedbackDataset` are allowed.")
```

Comment:
maybe "`dataset`" is not a correct quote here. I would say, provided object or something like that.